src/Service/Models/GraphQLFilterParsers.cs

```diff
@@ -113,13 +114,57 @@ public GQLFilterParser(ISqlMetadataProvider metadataProvider)
 
                     // Preserve the name value present in the filter.
+                    // In some cases, 'name' represents a relationship field on a source entity,
+                    // and not a field on a relationship target entity.
+                    // Additionally, 'name' may not be the same as the relationship's target entity because
+                    // runtime configuration supports overriding the entity name in the source entity configuration.
+                    // e.g.
+                    // comics (Source entity)  series (Target entity)
+                    // - id                    - id
+                    // - myseries [series]     - name
+                    // e.g. GraphQL request
+                    // {  comics ( filter: { myseries: { name: { eq: 'myName' } } } ) { <field selection> } }
                     string backingColumnName = name;
+
                     _metadataProvider.TryGetBackingColumn(queryStructure.EntityName, field: name, out string? resolvedBackingColumnName);
+
+                    // When runtime configuration defines relationship metadata,
+                    // an additional field on the entity representing GraphQL type
+                    // will exist. That relationship field does not represent a column in
+                    // the representative database table and the relationship field will not have an authorization
+                    // rule entry. Authorization is handled by permissions defined for the relationship's
+                    // target entity.
+                    bool relationshipField = true;
                     if (!string.IsNullOrWhiteSpace(resolvedBackingColumnName))
                     {
                         backingColumnName = resolvedBackingColumnName;
+                        relationshipField = false;
                     }
 
+                    // Only perform field (column) authorization when the field is not a relationship field.
+                    // Due to the recursive behavior of SqlExistsQueryStructure compilation, the column authorization
+                    // check only occurs when access to the column's owner entity is confirmed.
+                    if (!relationshipField)
+                    {
+                        string targetEntity = queryStructure.EntityName;
+
+                        bool columnAccessPermitted = queryStructure.AuthorizationResolver.AreColumnsAllowedForOperation(
+                            entityName: targetEntity,
+                            roleName: GetHttpContextFromMiddlewareContext(ctx).Request.Headers[CLIENT_ROLE_HEADER],
+                            operation: Config.Operation.Read,
+                            columns: new[] { name });
+
+                        if (!columnAccessPermitted)
+                        {
+                            throw new DataApiBuilderException(
+                                message: DataApiBuilderException.GRAPHQL_NESTEDFILTER_FIELD_AUTHZ_FAILURE,
```

Comment:
since this is not the NESTED FILTER case, does the constant used here need to be named differently?

Reply:
yep, thanks for calling out.

Reply:
updated to no longer say "nested"